profiles/profile.py

```diff
@@ -0,0 +1,156 @@
+"""
+Perform the image-based profiling pipeline to process data
+"""
+
+import os
+import pathlib
+import pandas as pd
+from pycytominer.aggregate import AggregateProfiles
+from pycytominer import (
+    aggregate,
+    annotate,
+    normalize,
+    feature_select,
+    audit,
+    cyto_utils,
+)
+
+from profile_utils import get_args
+
+# Load Command Line Arguments
+args = get_args()
+
+sql_file = args.sql_file
+batch = args.batch
+plate_name = args.plate_name
+platemap_file = args.platemap_file
+barcode_platemap_file = args.barcode_platemap_file
+moa_file = args.moa_file
+cell_count_dir = args.cell_count_dir
+output_dir = args.output_dir
+
+# Initialize profile processing
+os.makedirs(output_dir, exist_ok=True)
+os.makedirs(cell_count_dir, exist_ok=True)
+cell_id = "A549"
+aggregate_method = "median"
+norm_method = "mad_robustize"
+compression = "gzip"
+float_format = "%.5g"
+strata = ["Image_Metadata_Plate", "Image_Metadata_Well"]
+feature_select_ops = [
+    "drop_na_columns",
+    "variance_threshold",
+    "correlation_threshold",
+    "blacklist",
+]
+
+# Define external metadata to add to annotation
+moa_df = pd.read_csv(moa_file, sep="\t")
+barcode_platemap_df = pd.read_csv(barcode_platemap_file).query(
+    "Assay_Plate_Barcode == @plate_name"
```

Comment:
Ever since you introduced me to use f-string formatting syntax, I have been incorporating that to all my codes. As I saw you using f-strings extensively in the rest of the code, I wondered if you would like to write this query also as a f-string, though I don't see any issue in how it is currently written.

Reply:
f strings won't work inside a `pd.DataFrame().query()` call. The `@` specifically references an environment variable, in this case `plate_name`.  See [query doc](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.query.html).

Reply:
That's interesting. I had previously gotten f strings to work with `pd.DataFrame().query()` when I was querying for integers but it does look like it doesn't work for other data types. Anyway, there is no problem with your current implementation. Marking it as resolved.

Reply:
oh cool, I didn't know that the data type matters for this but I think it makes sense why. Good to know!